os/net/mac/tsch/tsch.c

```diff
@@ -274,8 +274,8 @@ keepalive_packet_sent(void *ptr, int status, int transmissions)
   LOG_INFO_LLADDR(packetbuf_addr(PACKETBUF_ADDR_RECEIVER));
   LOG_INFO_(", st %d-%d\n", status, transmissions);
 
-  /* We got no ack, try to recover by switching to the last neighbor we received an EB from */
-  if(status != MAC_TX_OK) {
+  /* We got no ack or collision, try to recover by switching to the last neighbor we received an EB from */
+  if(status == MAC_TX_NOACK || status == MAC_TX_COLLISION) {
```

Comment:
Ah no I meant the other way around. Collision are like errors: when they occur, we do not want to reschedule immediately. The test should be `if(status == MAC_TX_NOACK)`

Reply:
Indeed, my bad. It is corrected now